### PR TITLE
Added trending section with trending hashtags

### PR DESCRIPTION
**Pull request by devotaaabel**
- Did some setup for trending links and trending statuses as well
- TODO setup retry
- TODO add tabs, statuses and links

### CODE REVIEW

File: core/database/src/main/java/social/firefly/core/database/SocialDatabase.kt [hunk 1]
```diff
@@ -79,6 +85,9 @@ import social.firefly.core.database.model.entities.statusCollections.SearchedSta
         MainNotification::class,
         MentionListNotification::class,
         FollowListNotification::class,
+        TrendingLink::class,
+        TrendingStatus::class,
+        TrendingHashTag::class,
     ],
     version = 1,
```
Comment (JohnOberhauser) [inline core/database/src/main/java/social/firefly/core/database/SocialDatabase.kt]: We need to either increase the version number here, undo the changes to the `1.json` schema, and generate a `2.json` schema

or

Enable destructive migrations in room and make sure it doesn't crash if we don't update the schema version

File: core/database/src/main/java/social/firefly/core/database/model/entities/statusCollections/TrendingStatus.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+)
+data class TrendingStatus(
+    @PrimaryKey
+    val statusId: String,
+    val position: Int,
+)
+
+data class TrendingStatusWrapper(
+    @Embedded
+    val searchedStatus: SearchedStatus,
```
Comment (JohnOberhauser) [inline core/database/src/main/java/social/firefly/core/database/model/entities/statusCollections/TrendingStatus.kt]: this should be `TrendingStatus`

File: core/network/mastodon/src/main/java/social/firefly/core/network/mastodon/TrendsApi.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    @GET("/api/v1/trends/tags")
+    suspend fun getTrendingTags(
+        @Query("limit") limit: Int? = null,
+        @Query("offset") offset: Int? = null,
+    ): List<NetworkHashTag>
+
+    @GET("/api/v1/trends/statuses")
+    suspend fun getTrendingStatuses(
+        limit: Int? = null,
+        offset: Int? = null,
```
Comment (JohnOberhauser) [inline core/network/mastodon/src/main/java/social/firefly/core/network/mastodon/TrendsApi.kt]: I think these need `@Query` annotations

File: core/database/src/main/java/social/firefly/core/database/SocialDatabase.kt (hunk 1 above)
Comment (devotaaabel) [reply core/database/src/main/java/social/firefly/core/database/SocialDatabase.kt]: I added destructive migrations but didn't increment the db.  I thought we were going to just keep it at v1 and require clearing data until an official release?

Comment (JohnOberhauser) [reply core/database/src/main/java/social/firefly/core/database/SocialDatabase.kt]: I'm fine with destructive migrations until we release, but it would be nice to increment the version so when my app auto-updates through obtainium, it doesn't just crash when I open it.

Comment (JohnOberhauser) [reply core/database/src/main/java/social/firefly/core/database/SocialDatabase.kt]: We can roll back to version 1 before a release release

**Review by JohnOberhauser — APPROVED** [review]
